src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,43 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
-
+        int supplySum = 0;
+        int buySum = 0;
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");
+                if (parts.length != 2) {
+                    System.out.println("Невірний формат рядка: " + line);
+                    continue;
+                }
+                String operation = parts[0];
+                int amount = Integer.parseInt(parts[1]);
+                if (operation.equals("supply")) {
+                    supplySum += amount;
+                } else if (operation.equals("buy")) {
+                    buySum += amount;
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from file: " + fromFileName, e);
+        }
+        int result = supplySum - buySum;
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+            writer.write("supply," + supplySum);
+            writer.newLine();
+            writer.write("buy," + buySum);
+            writer.newLine();
+            writer.write("result," + result);
```

Comment:
Using hardcoded strings like "supply", "buy", and "result" can make the code harder to maintain. Consider defining these as constants at the class level.